BUG: io/scipy.sparse.csgraph: refguide check failure in main

### Describe your issue.

The refguide check is failing in CI (e.g. gh-21167, gh-21201) with the errors below.

### Reproducing Code Example

```python
Requirement already satisfied: matplotlib in /home/circleci/.pyenv/versions/3.11.9/lib/python3.11/site-packages (3.9.1)
Collecting hypothesis
  Downloading hypothesis-6.108.4-py3-none-any.whl.metadata (6.3 kB)
Requirement already satisfied: contourpy>=1.0.1 in /home/circleci/.pyenv/versions/3.11.9/lib/python3.11/site-packages (from matplotlib) (1.2.1)
Requirement already satisfied: cycler>=0.10 in /home/circleci/.pyenv/versions/3.11.9/lib/python3.11/site-packages (from matplotlib) (0.12.1)
Requirement already satisfied: fonttools>=4.22.0 in /home/circleci/.pyenv/versions/3.11.9/lib/python3.11/site-packages (from matplotlib) (4.53.1)
Requirement already satisfied: kiwisolver>=1.3.1 in /home/circleci/.pyenv/versions/3.11.9/lib/python3.11/site-packages (from matplotlib) (1.4.5)
Requirement already satisfied: numpy>=1.23 in /home/circleci/.pyenv/versions/3.11.9/lib/python3.11/site-packages (from matplotlib) (2.0.1)
Requirement already satisfied: packaging>=20.0 in /home/circleci/.local/lib/python3.11/site-packages (from matplotlib) (24.0)
Requirement already satisfied: pillow>=8 in /home/circleci/.pyenv/versions/3.11.9/lib/python3.11/site-packages (from matplotlib) (10.4.0)
Requirement already satisfied: pyparsing>=2.3.1 in /home/circleci/.pyenv/versions/3.11.9/lib/python3.11/site-packages (from matplotlib) (3.1.2)
Requirement already satisfied: python-dateutil>=2.7 in /home/circleci/.pyenv/versions/3.11.9/lib/python3.11/site-packages (from matplotlib) (2.9.0.post0)
Requirement already satisfied: attrs>=22.2.0 in /home/circleci/.pyenv/versions/3.11.9/lib/python3.11/site-packages (from hypothesis) (23.2.0)
Collecting sortedcontainers<3.0.0,>=2.1.0 (from hypothesis)
  Downloading sortedcontainers-2.4.0-py2.py3-none-any.whl.metadata (10 kB)
Requirement already satisfied: six>=1.5 in /home/circleci/.pyenv/versions/3.11.9/lib/python3.11/site-packages (from python-dateutil>=2.7->matplotlib) (1.16.0)
Downloading hypothesis-6.108.4-py3-none-any.whl (465 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 465.2/465.2 kB 14.6 MB/s eta 0:00:00

Downloading sortedcontainers-2.4.0-py2.py3-none-any.whl (29 kB)
Installing collected packages: sortedcontainers, hypothesis
Successfully installed hypothesis-6.108.4 sortedcontainers-2.4.0

[notice] A new release of pip is available: 24.0 -> 24.1.2
[notice] To update, run: pip install --upgrade pip
Collecting scipy-doctest
  Downloading scipy_doctest-1.3-py3-none-any.whl.metadata (14 kB)
Requirement already satisfied: numpy>=1.19.5 in /home/circleci/.pyenv/versions/3.11.9/lib/python3.11/site-packages (from scipy-doctest) (2.0.1)
Requirement already satisfied: pytest in /home/circleci/.pyenv/versions/3.11.9/lib/python3.11/site-packages (from scipy-doctest) (8.3.1)
Requirement already satisfied: iniconfig in /home/circleci/.pyenv/versions/3.11.9/lib/python3.11/site-packages (from pytest->scipy-doctest) (2.0.0)
Requirement already satisfied: packaging in /home/circleci/.local/lib/python3.11/site-packages (from pytest->scipy-doctest) (24.0)
Requirement already satisfied: pluggy<2,>=1.5 in /home/circleci/.pyenv/versions/3.11.9/lib/python3.11/site-packages (from pytest->scipy-doctest) (1.5.0)
Downloading scipy_doctest-1.3-py3-none-any.whl (60 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 60.1/60.1 kB 3.5 MB/s eta 0:00:00

Installing collected packages: scipy-doctest
Successfully installed scipy-doctest-1.3

[notice] A new release of pip is available: 24.0 -> 24.1.2
[notice] To update, run: pip install --upgrade pip
💻  ninja -C /home/circleci/repo/build -j2
ninja: Entering directory `/home/circleci/repo/build'
[1/41] Generating scipy/linalg/_dummy_cython_blas.pxd with a custom command
[2/2] Generating scipy/generate-version with a custom command
Build OK
💻  meson install -C build --only-changed --tags 
runtime,python-runtime,tests,devel
Installing, see meson-install.log...
Installation OK
SciPy from development installed path at: /home/circleci/repo/build-install/lib/python3.11/site-packages
Running tests for scipy version:1.15.0.dev0+1229.bdd3a28, installed at:/home/circleci/repo/build-install/lib/python3.11/site-packages/scipy
============================= test session starts ==============================
platform linux -- Python 3.11.9, pytest-8.3.1, pluggy-1.5.0
rootdir: /home/circleci/repo
configfile: pytest.ini
plugins: anyio-4.4.0, hypothesis-6.108.4, scipy_doctest-1.3, typeguard-4.3.0
collected 2519 items                                                           

scipy/cluster/hierarchy.py ...............................               [  2%]
scipy/cluster/vq.py ....                                                 [  2%]
scipy/constants/__init__.py ...............                              [  3%]
scipy/datasets/__init__.py ....                                          [  4%]
scipy/differentiate/__init__.py ..                                       [  4%]
scipy/fft/__init__.py ..................................                 [  6%]
scipy/fftpack/__init__.py ................                               [  8%]
scipy/integrate/__init__.py ..................                           [  9%]
scipy/interpolate/__init__.py .......................................... [ 12%]
.........                                                                [ 13%]
scipy/io/__init__.py .FF........                                         [ 14%]
scipy/io/arff/__init__.py .                                              [ 14%]
scipy/io/matlab/__init__.py ...                                          [ 14%]
scipy/io/wavfile.py ..                                                   [ 14%]
scipy/linalg/__init__.py ............................................... [ 18%]
..................................................                       [ 21%]
scipy/linalg/blas.py ..                                                  [ 22%]
scipy/linalg/interpolative.py .                                          [ 22%]
scipy/linalg/lapack.py .                                                 [ 22%]
scipy/ndimage/__init__.py .............................................. [ 25%]
.....................                                                    [ 27%]
scipy/odr/__init__.py .....                                              [ 27%]
scipy/optimize/__init__.py ............................................. [ 31%]
........                                                                 [ 31%]
scipy/optimize/elementwise.py ....                                       [ 32%]
scipy/signal/__init__.py ............................................... [ 35%]
........................................................................ [ 41%]
..............                                                           [ 42%]
scipy/signal/windows/__init__.py ..........................              [ 44%]
scipy/sparse/__init__.py ..........................................      [ 47%]
scipy/sparse/csgraph/__init__.py F.FFFFFFFF...F...FFF......              [ 49%]
scipy/sparse/linalg/__init__.py ...............................          [ 51%]
scipy/spatial/__init__.py ..........................                     [ 53%]
scipy/spatial/distance.py .............................                  [ 55%]
scipy/spatial/transform/__init__.py ..........................           [ 57%]
scipy/special/__init__.py .............................................. [ 61%]
........................................................................ [ 66%]
........................................................................ [ 72%]
..........................                                               [ 74%]
scipy/stats/__init__.py ................................................ [ 77%]
........................................................................ [ 83%]
........................................................................ [ 88%]
........................................................................ [ 94%]
.....................                                                    [ 95%]
scipy/stats/contingency.py .......                                       [ 96%]
scipy/stats/mstats.py ...........................                        [ 98%]
scipy/stats/qmc.py ...........                                           [ 99%]
scipy/stats/sampling.py ...........                                      [100%]

=================================== FAILURES ===================================
__________________________ [doctest] scipy.io.hb_read __________________________
491 
492     Examples
493     --------
494     We can read and write a harwell-boeing format file:
495 
496     >>> from scipy.io import hb_read, hb_write
497     >>> from scipy.sparse import csr_array, eye
498     >>> data = csr_array(eye(3))  # create a sparse array
499     >>> hb_write("data.hb", data)  # write a hb file
500     >>> print(hb_read("data.hb"))  # read a hb file
Differences (unified diff with -expected +actual):
    @@ -1,3 +1,6 @@
    -(np.int32(0), np.int32(0))  1.0
    -(np.int32(1), np.int32(1))  1.0
    -(np.int32(2), np.int32(2))  1.0
    +<Compressed Sparse Column sparse matrix of dtype 'float64'
    +   with 3 stored elements and shape (3, 3)>
    +  Coords   Values
    +  (0, 0)   1.0
    +  (1, 1)   1.0
    +  (2, 2)   1.0

/home/circleci/repo/build-install/lib/python3.11/site-packages/scipy/io/_harwell_boeing/hb.py:500: DocTestFailure
_________________________ [doctest] scipy.io.hb_write __________________________
541 
542     Examples
543     --------
544     We can read and write a harwell-boeing format file:
545 
546     >>> from scipy.io import hb_read, hb_write
547     >>> from scipy.sparse import csr_array, eye
548     >>> data = csr_array(eye(3))  # create a sparse array
549     >>> hb_write("data.hb", data)  # write a hb file
550     >>> print(hb_read("data.hb"))  # read a hb file
Differences (unified diff with -expected +actual):
    @@ -1,3 +1,6 @@
    -(np.int32(0), np.int32(0))  1.0
    -(np.int32(1), np.int32(1))  1.0
    -(np.int32(2), np.int32(2))  1.0
    +<Compressed Sparse Column sparse matrix of dtype 'float64'
    +   with 3 stored elements and shape (3, 3)>
    +  Coords   Values
    +  (0, 0)   1.0
    +  (1, 1)   1.0
    +  (2, 2)   1.0

/home/circleci/repo/build-install/lib/python3.11/site-packages/scipy/io/_harwell_boeing/hb.py:550: DocTestFailure
_____________ [doctest] scipy.sparse.csgraph.connected_components ______________
EXAMPLE LOCATION UNKNOWN, not showing all tests of that example
??? >>> print(graph)
Differences (unified diff with -expected +actual):
    @@ -1,4 +1,7 @@
    -  (np.int32(0), np.int32(1))        1
    -  (np.int32(0), np.int32(2))        1
    -  (np.int32(1), np.int32(2))        1
    -  (np.int32(3), np.int32(4))        1
    +<Compressed Sparse Row sparse matrix of dtype 'int64'
    +   with 4 stored elements and shape (5, 5)>
    +  Coords   Values
    +  (0, 1)   1
    +  (0, 2)   1
    +  (1, 2)   1
    +  (3, 4)   1

/home/circleci/repo/build-install/lib/python3.11/site-packages/scipy/sparse/csgraph/_traversal.cpython-311-x86_64-linux-gnu.so:None: DocTestFailure
_________________ [doctest] scipy.sparse.csgraph.shortest_path _________________
EXAMPLE LOCATION UNKNOWN, not showing all tests of that example
??? >>> print(graph)
Differences (unified diff with -expected +actual):
    @@ -1,5 +1,8 @@
    -  (np.int32(0), np.int32(1))        1
    -  (np.int32(0), np.int32(2))        2
    -  (np.int32(1), np.int32(3))        1
    -  (np.int32(2), np.int32(0))        2
    -  (np.int32(2), np.int32(3))        3
    +<Compressed Sparse Row sparse matrix of dtype 'int64'
    +   with 5 stored elements and shape (4, 4)>
    +  Coords   Values
    +  (0, 1)   1
    +  (0, 2)   2
    +  (1, 3)   1
    +  (2, 0)   2
    +  (2, 3)   3

/home/circleci/repo/build-install/lib/python3.11/site-packages/scipy/sparse/csgraph/_shortest_path.cpython-311-x86_64-linux-gnu.so:None: DocTestFailure
________________ [doctest] scipy.sparse.csgraph.floyd_warshall _________________
EXAMPLE LOCATION UNKNOWN, not showing all tests of that example
??? >>> print(graph)
Differences (unified diff with -expected +actual):
    @@ -1,5 +1,8 @@
    -  (np.int32(0), np.int32(1))        1
    -  (np.int32(0), np.int32(2))        2
    -  (np.int32(1), np.int32(3))        1
    -  (np.int32(2), np.int32(0))        2
    -  (np.int32(2), np.int32(3))        3
    +<Compressed Sparse Row sparse matrix of dtype 'int64'
    +   with 5 stored elements and shape (4, 4)>
    +  Coords   Values
    +  (0, 1)   1
    +  (0, 2)   2
    +  (1, 3)   1
    +  (2, 0)   2
    +  (2, 3)   3

/home/circleci/repo/build-install/lib/python3.11/site-packages/scipy/sparse/csgraph/_shortest_path.cpython-311-x86_64-linux-gnu.so:None: DocTestFailure
___________________ [doctest] scipy.sparse.csgraph.dijkstra ____________________
EXAMPLE LOCATION UNKNOWN, not showing all tests of that example
??? >>> print(graph)
Differences (unified diff with -expected +actual):
    @@ -1,4 +1,7 @@
    -  (np.int32(0), np.int32(1))        1
    -  (np.int32(0), np.int32(2))        2
    -  (np.int32(1), np.int32(3))        1
    -  (np.int32(2), np.int32(3))        3
    +<Compressed Sparse Row sparse matrix of dtype 'int64'
    +   with 4 stored elements and shape (4, 4)>
    +  Coords   Values
    +  (0, 1)   1
    +  (0, 2)   2
    +  (1, 3)   1
    +  (2, 3)   3

/home/circleci/repo/build-install/lib/python3.11/site-packages/scipy/sparse/csgraph/_shortest_path.cpython-311-x86_64-linux-gnu.so:None: DocTestFailure
_________________ [doctest] scipy.sparse.csgraph.bellman_ford __________________
EXAMPLE LOCATION UNKNOWN, not showing all tests of that example
??? >>> print(graph)
Differences (unified diff with -expected +actual):
    @@ -1,5 +1,8 @@
    -  (np.int32(0), np.int32(1))        1
    -  (np.int32(0), np.int32(2))        2
    -  (np.int32(1), np.int32(3))        1
    -  (np.int32(2), np.int32(0))        2
    -  (np.int32(2), np.int32(3))        3
    +<Compressed Sparse Row sparse matrix of dtype 'int64'
    +   with 5 stored elements and shape (4, 4)>
    +  Coords   Values
    +  (0, 1)   1
    +  (0, 2)   2
    +  (1, 3)   1
    +  (2, 0)   2
    +  (2, 3)   3

/home/circleci/repo/build-install/lib/python3.11/site-packages/scipy/sparse/csgraph/_shortest_path.cpython-311-x86_64-linux-gnu.so:None: DocTestFailure
____________________ [doctest] scipy.sparse.csgraph.johnson ____________________
EXAMPLE LOCATION UNKNOWN, not showing all tests of that example
??? >>> print(graph)
Differences (unified diff with -expected +actual):
    @@ -1,5 +1,8 @@
    -  (np.int32(0), np.int32(1))        1
    -  (np.int32(0), np.int32(2))        2
    -  (np.int32(1), np.int32(3))        1
    -  (np.int32(2), np.int32(0))        2
    -  (np.int32(2), np.int32(3))        3
    +<Compressed Sparse Row sparse matrix of dtype 'int64'
    +   with 5 stored elements and shape (4, 4)>
    +  Coords   Values
    +  (0, 1)   1
    +  (0, 2)   2
    +  (1, 3)   1
    +  (2, 0)   2
    +  (2, 3)   3

/home/circleci/repo/build-install/lib/python3.11/site-packages/scipy/sparse/csgraph/_shortest_path.cpython-311-x86_64-linux-gnu.so:None: DocTestFailure
______________________ [doctest] scipy.sparse.csgraph.yen ______________________
EXAMPLE LOCATION UNKNOWN, not showing all tests of that example
??? >>> print(graph)
Differences (unified diff with -expected +actual):
    @@ -1,5 +1,8 @@
    -  (np.int32(0), np.int32(1))        1
    -  (np.int32(0), np.int32(2))        2
    -  (np.int32(1), np.int32(3))        1
    -  (np.int32(2), np.int32(0))        2
    -  (np.int32(2), np.int32(3))        3
    +<Compressed Sparse Row sparse matrix of dtype 'int64'
    +   with 5 stored elements and shape (4, 4)>
    +  Coords   Values
    +  (0, 1)   1
    +  (0, 2)   2
    +  (1, 3)   1
    +  (2, 0)   2
    +  (2, 3)   3

/home/circleci/repo/build-install/lib/python3.11/site-packages/scipy/sparse/csgraph/_shortest_path.cpython-311-x86_64-linux-gnu.so:None: DocTestFailure
______________ [doctest] scipy.sparse.csgraph.breadth_first_order ______________
EXAMPLE LOCATION UNKNOWN, not showing all tests of that example
??? >>> print(graph)
Differences (unified diff with -expected +actual):
    @@ -1,5 +1,8 @@
    -  (np.int32(0), np.int32(1))        1
    -  (np.int32(0), np.int32(2))        2
    -  (np.int32(1), np.int32(3))        1
    -  (np.int32(2), np.int32(0))        2
    -  (np.int32(2), np.int32(3))        3
    +<Compressed Sparse Row sparse matrix of dtype 'int64'
    +   with 5 stored elements and shape (4, 4)>
    +  Coords   Values
    +  (0, 1)   1
    +  (0, 2)   2
    +  (1, 3)   1
    +  (2, 0)   2
    +  (2, 3)   3

/home/circleci/repo/build-install/lib/python3.11/site-packages/scipy/sparse/csgraph/_traversal.cpython-311-x86_64-linux-gnu.so:None: DocTestFailure
_______________ [doctest] scipy.sparse.csgraph.depth_first_order _______________
EXAMPLE LOCATION UNKNOWN, not showing all tests of that example
??? >>> print(graph)
Differences (unified diff with -expected +actual):
    @@ -1,5 +1,8 @@
    -(np.int32(0), np.int32(1))  1
    -(np.int32(0), np.int32(2))  2
    -(np.int32(1), np.int32(3))  1
    -(np.int32(2), np.int32(0))  2
    -(np.int32(2), np.int32(3))  3
    +<Compressed Sparse Row sparse matrix of dtype 'int64'
    +   with 5 stored elements and shape (4, 4)>
    +  Coords   Values
    +  (0, 1)   1
    +  (0, 2)   2
    +  (1, 3)   1
    +  (2, 0)   2
    +  (2, 3)   3

/home/circleci/repo/build-install/lib/python3.11/site-packages/scipy/sparse/csgraph/_traversal.cpython-311-x86_64-linux-gnu.so:None: DocTestFailure
_____________ [doctest] scipy.sparse.csgraph.reverse_cuthill_mckee _____________
EXAMPLE LOCATION UNKNOWN, not showing all tests of that example
??? >>> print(graph)
Differences (unified diff with -expected +actual):
    @@ -1,5 +1,8 @@
    -  (np.int32(0), np.int32(1))        1
    -  (np.int32(0), np.int32(2))        2
    -  (np.int32(1), np.int32(3))        1
    -  (np.int32(2), np.int32(0))        2
    -  (np.int32(2), np.int32(3))        3
    +<Compressed Sparse Row sparse matrix of dtype 'int64'
    +   with 5 stored elements and shape (4, 4)>
    +  Coords   Values
    +  (0, 1)   1
    +  (0, 2)   2
    +  (1, 3)   1
    +  (2, 0)   2
    +  (2, 3)   3

/home/circleci/repo/build-install/lib/python3.11/site-packages/scipy/sparse/csgraph/_reordering.cpython-311-x86_64-linux-gnu.so:None: DocTestFailure
________________ [doctest] scipy.sparse.csgraph.structural_rank ________________
EXAMPLE LOCATION UNKNOWN, not showing all tests of that example
??? >>> print(graph)
Differences (unified diff with -expected +actual):
    @@ -1,8 +1,11 @@
    -  (np.int32(0), np.int32(1))        1
    -  (np.int32(0), np.int32(2))        2
    -  (np.int32(1), np.int32(0))        1
    -  (np.int32(1), np.int32(3))        1
    -  (np.int32(2), np.int32(0))        2
    -  (np.int32(2), np.int32(3))        3
    -  (np.int32(3), np.int32(1))        1
    -  (np.int32(3), np.int32(2))        3
    +<Compressed Sparse Row sparse matrix of dtype 'int64'
    +   with 8 stored elements and shape (4, 4)>
    +  Coords   Values
    +  (0, 1)   1
    +  (0, 2)   2
    +  (1, 0)   1
    +  (1, 3)   1
    +  (2, 0)   2
    +  (2, 3)   3
    +  (3, 1)   1
    +  (3, 2)   3

/home/circleci/repo/build-install/lib/python3.11/site-packages/scipy/sparse/csgraph/_reordering.cpython-311-x86_64-linux-gnu.so:None: DocTestFailure
_____________ [doctest] scipy.sparse.csgraph.construct_dist_matrix _____________
EXAMPLE LOCATION UNKNOWN, not showing all tests of that example
??? >>> print(graph)
Differences (unified diff with -expected +actual):
    @@ -1,4 +1,7 @@
    -   (np.int32(0), np.int32(1))       1
    -   (np.int32(0), np.int32(2))       2
    -   (np.int32(1), np.int32(3))       1
    -   (np.int32(2), np.int32(3))       3
    +<Compressed Sparse Row sparse matrix of dtype 'int64'
    +   with 4 stored elements and shape (4, 4)>
    +  Coords   Values
    +  (0, 1)   1
    +  (0, 2)   2
    +  (1, 3)   1
    +  (2, 3)   3

/home/circleci/repo/build-install/lib/python3.11/site-packages/scipy/sparse/csgraph/_tools.cpython-311-x86_64-linux-gnu.so:None: DocTestFailure
_______________ [doctest] scipy.sparse.csgraph.reconstruct_path ________________
EXAMPLE LOCATION UNKNOWN, not showing all tests of that example
??? >>> print(graph)
Differences (unified diff with -expected +actual):
    @@ -1,4 +1,7 @@
    -   (np.int32(0), np.int32(1))       1
    -   (np.int32(0), np.int32(2))       2
    -   (np.int32(1), np.int32(3))       1
    -   (np.int32(2), np.int32(3))       3
    +<Compressed Sparse Row sparse matrix of dtype 'int64'
    +   with 4 stored elements and shape (4, 4)>
    +  Coords   Values
    +  (0, 1)   1
    +  (0, 2)   2
    +  (1, 3)   1
    +  (2, 3)   3

/home/circleci/repo/build-install/lib/python3.11/site-packages/scipy/sparse/csgraph/_tools.cpython-311-x86_64-linux-gnu.so:None: DocTestFailure
=========================== short test summary info ============================
FAILED scipy/io/__init__.py::scipy.io.hb_read
FAILED scipy/io/__init__.py::scipy.io.hb_write
FAILED scipy/sparse/csgraph/__init__.py::scipy.sparse.csgraph.connected_components
FAILED scipy/sparse/csgraph/__init__.py::scipy.sparse.csgraph.shortest_path
FAILED scipy/sparse/csgraph/__init__.py::scipy.sparse.csgraph.floyd_warshall
FAILED scipy/sparse/csgraph/__init__.py::scipy.sparse.csgraph.dijkstra
FAILED scipy/sparse/csgraph/__init__.py::scipy.sparse.csgraph.bellman_ford
FAILED scipy/sparse/csgraph/__init__.py::scipy.sparse.csgraph.johnson
FAILED scipy/sparse/csgraph/__init__.py::scipy.sparse.csgraph.yen
FAILED scipy/sparse/csgraph/__init__.py::scipy.sparse.csgraph.breadth_first_order
FAILED scipy/sparse/csgraph/__init__.py::scipy.sparse.csgraph.depth_first_order
FAILED scipy/sparse/csgraph/__init__.py::scipy.sparse.csgraph.reverse_cuthill_mckee
FAILED scipy/sparse/csgraph/__init__.py::scipy.sparse.csgraph.structural_rank
FAILED scipy/sparse/csgraph/__init__.py::scipy.sparse.csgraph.construct_dist_matrix
FAILED scipy/sparse/csgraph/__init__.py::scipy.sparse.csgraph.reconstruct_path
================= 15 failed, 1303 passed in 1804.52s (0:30:04) =================
```


### Error message

```shell
-
```


### SciPy/NumPy/Python version and system information

```shell
-
```

--- a/scipy/io/_harwell_boeing/hb.py
+++ b/scipy/io/_harwell_boeing/hb.py
@@ -499,11 +499,11 @@ def hb_read(path_or_open_file):
     >>> hb_write("data.hb", data)  # write a hb file
     >>> print(hb_read("data.hb"))  # read a hb file
     <Compressed Sparse Column sparse matrix of dtype 'float64'
-    	with 3 stored elements and shape (3, 3)>
-    	Coords	Values
-    	(0, 0)	1.0
-    	(1, 1)	1.0
-    	(2, 2)	1.0
+        with 3 stored elements and shape (3, 3)>
+        Coords	Values
+        (0, 0)	1.0
+        (1, 1)	1.0
+        (2, 2)	1.0
     """
     def _get_matrix(fid):
         hb = HBFile(fid)
@@ -552,11 +552,11 @@ def hb_write(path_or_open_file, m, hb_info=None):
     >>> hb_write("data.hb", data)  # write a hb file
     >>> print(hb_read("data.hb"))  # read a hb file
     <Compressed Sparse Column sparse matrix of dtype 'float64'
-    	with 3 stored elements and shape (3, 3)>
-    	Coords	Values
-    	(0, 0)	1.0
-    	(1, 1)	1.0
-    	(2, 2)	1.0
+        with 3 stored elements and shape (3, 3)>
+        Coords	Values
+        (0, 0)	1.0
+        (1, 1)	1.0
+        (2, 2)	1.0
     """
     m = m.tocsc(copy=False)
 

--- a/scipy/io/_harwell_boeing/hb.py
+++ b/scipy/io/_harwell_boeing/hb.py
@@ -498,9 +498,12 @@ def hb_read(path_or_open_file):
     >>> data = csr_array(eye(3))  # create a sparse array
     >>> hb_write("data.hb", data)  # write a hb file
     >>> print(hb_read("data.hb"))  # read a hb file
-    (np.int32(0), np.int32(0))	1.0
-    (np.int32(1), np.int32(1))	1.0
-    (np.int32(2), np.int32(2))	1.0
+    <Compressed Sparse Column sparse matrix of dtype 'float64'
+    	with 3 stored elements and shape (3, 3)>
+    	Coords	Values
+    	(0, 0)	1.0
+    	(1, 1)	1.0
+    	(2, 2)	1.0
     """
     def _get_matrix(fid):
         hb = HBFile(fid)
@@ -548,9 +551,12 @@ def hb_write(path_or_open_file, m, hb_info=None):
     >>> data = csr_array(eye(3))  # create a sparse array
     >>> hb_write("data.hb", data)  # write a hb file
     >>> print(hb_read("data.hb"))  # read a hb file
-    (np.int32(0), np.int32(0))	1.0
-    (np.int32(1), np.int32(1))	1.0
-    (np.int32(2), np.int32(2))	1.0
+    <Compressed Sparse Column sparse matrix of dtype 'float64'
+    	with 3 stored elements and shape (3, 3)>
+    	Coords	Values
+    	(0, 0)	1.0
+    	(1, 1)	1.0
+    	(2, 2)	1.0
     """
     m = m.tocsc(copy=False)
 

--- a/scipy/sparse/csgraph/_reordering.pyx
+++ b/scipy/sparse/csgraph/_reordering.pyx
@@ -60,13 +60,13 @@ def reverse_cuthill_mckee(graph, symmetric_mode=False):
     >>> graph = csr_matrix(graph)
     >>> print(graph)
     <Compressed Sparse Row sparse matrix of dtype 'int64'
-    	with 5 stored elements and shape (4, 4)>
-    	Coords	Values
-    	(0, 1)	1
-    	(0, 2)	2
-    	(1, 3)	1
-    	(2, 0)	2
-    	(2, 3)	3
+        with 5 stored elements and shape (4, 4)>
+        Coords	Values
+        (0, 1)	1
+        (0, 2)	2
+        (1, 3)	1
+        (2, 0)	2
+        (2, 3)	3
 
     >>> reverse_cuthill_mckee(graph)
     array([3, 2, 1, 0], dtype=int32)
@@ -219,16 +219,16 @@ def structural_rank(graph):
     >>> graph = csr_matrix(graph)
     >>> print(graph)
     <Compressed Sparse Row sparse matrix of dtype 'int64'
-    	with 8 stored elements and shape (4, 4)>
-    	Coords	Values
-    	(0, 1)	1
-    	(0, 2)	2
-    	(1, 0)	1
-    	(1, 3)	1
-    	(2, 0)	2
-    	(2, 3)	3
-    	(3, 1)	1
-    	(3, 2)	3
+        with 8 stored elements and shape (4, 4)>
+        Coords	Values
+        (0, 1)	1
+        (0, 2)	2
+        (1, 0)	1
+        (1, 3)	1
+        (2, 0)	2
+        (2, 3)	3
+        (3, 1)	1
+        (3, 2)	3
 
     >>> structural_rank(graph)
     4

--- a/scipy/sparse/csgraph/_reordering.pyx
+++ b/scipy/sparse/csgraph/_reordering.pyx
@@ -59,11 +59,14 @@ def reverse_cuthill_mckee(graph, symmetric_mode=False):
     ... ]
     >>> graph = csr_matrix(graph)
     >>> print(graph)
-      (np.int32(0), np.int32(1))	1
-      (np.int32(0), np.int32(2))	2
-      (np.int32(1), np.int32(3))	1
-      (np.int32(2), np.int32(0))	2
-      (np.int32(2), np.int32(3))	3
+    <Compressed Sparse Row sparse matrix of dtype 'int64'
+    	with 5 stored elements and shape (4, 4)>
+    	Coords	Values
+    	(0, 1)	1
+    	(0, 2)	2
+    	(1, 3)	1
+    	(2, 0)	2
+    	(2, 3)	3
 
     >>> reverse_cuthill_mckee(graph)
     array([3, 2, 1, 0], dtype=int32)
@@ -215,14 +218,17 @@ def structural_rank(graph):
     ... ]
     >>> graph = csr_matrix(graph)
     >>> print(graph)
-      (np.int32(0), np.int32(1))	1
-      (np.int32(0), np.int32(2))	2
-      (np.int32(1), np.int32(0))	1
-      (np.int32(1), np.int32(3))	1
-      (np.int32(2), np.int32(0))	2
-      (np.int32(2), np.int32(3))	3
-      (np.int32(3), np.int32(1))	1
-      (np.int32(3), np.int32(2))	3
+    <Compressed Sparse Row sparse matrix of dtype 'int64'
+    	with 8 stored elements and shape (4, 4)>
+    	Coords	Values
+    	(0, 1)	1
+    	(0, 2)	2
+    	(1, 0)	1
+    	(1, 3)	1
+    	(2, 0)	2
+    	(2, 3)	3
+    	(3, 1)	1
+    	(3, 2)	3
 
     >>> structural_rank(graph)
     4

--- a/scipy/sparse/csgraph/_shortest_path.pyx
+++ b/scipy/sparse/csgraph/_shortest_path.pyx
@@ -148,11 +148,14 @@ def shortest_path(csgraph, method='auto',
     ... ]
     >>> graph = csr_matrix(graph)
     >>> print(graph)
-      (np.int32(0), np.int32(1))	1
-      (np.int32(0), np.int32(2))	2
-      (np.int32(1), np.int32(3))	1
-      (np.int32(2), np.int32(0))	2
-      (np.int32(2), np.int32(3))	3
+    <Compressed Sparse Row sparse matrix of dtype 'int64'
+    	with 5 stored elements and shape (4, 4)>
+    	Coords	Values
+    	(0, 1)	1
+    	(0, 2)	2
+    	(1, 3)	1
+    	(2, 0)	2
+    	(2, 3)	3
 
     >>> dist_matrix, predecessors = shortest_path(csgraph=graph, directed=False, indices=0, return_predecessors=True)
     >>> dist_matrix
@@ -296,11 +299,14 @@ def floyd_warshall(csgraph, directed=True,
     ... ]
     >>> graph = csr_matrix(graph)
     >>> print(graph)
-      (np.int32(0), np.int32(1))	1
-      (np.int32(0), np.int32(2))	2
-      (np.int32(1), np.int32(3))	1
-      (np.int32(2), np.int32(0))	2
-      (np.int32(2), np.int32(3))	3
+    <Compressed Sparse Row sparse matrix of dtype 'int64'
+    	with 5 stored elements and shape (4, 4)>
+    	Coords	Values
+    	(0, 1)	1
+    	(0, 2)	2
+    	(1, 3)	1
+    	(2, 0)	2
+    	(2, 3)	3
 
     >>> dist_matrix, predecessors = floyd_warshall(csgraph=graph, directed=False, return_predecessors=True)
     >>> dist_matrix
@@ -522,10 +528,13 @@ def dijkstra(csgraph, directed=True, indices=None,
     ... ]
     >>> graph = csr_matrix(graph)
     >>> print(graph)
-      (np.int32(0), np.int32(1))	1
-      (np.int32(0), np.int32(2))	2
-      (np.int32(1), np.int32(3))	1
-      (np.int32(2), np.int32(3))	3
+    <Compressed Sparse Row sparse matrix of dtype 'int64'
+    	with 4 stored elements and shape (4, 4)>
+    	Coords	Values
+    	(0, 1)	1
+    	(0, 2)	2
+    	(1, 3)	1
+    	(2, 3)	3
 
     >>> dist_matrix, predecessors = dijkstra(csgraph=graph, directed=False, indices=0, return_predecessors=True)
     >>> dist_matrix
@@ -1016,11 +1025,14 @@ def bellman_ford(csgraph, directed=True, indices=None,
     ... ]
     >>> graph = csr_matrix(graph)
     >>> print(graph)
-      (np.int32(0), np.int32(1))	1
-      (np.int32(0), np.int32(2))	2
-      (np.int32(1), np.int32(3))	1
-      (np.int32(2), np.int32(0))	2
-      (np.int32(2), np.int32(3))	3
+    <Compressed Sparse Row sparse matrix of dtype 'int64'
+    	with 5 stored elements and shape (4, 4)>
+    	Coords	Values
+    	(0, 1)	1
+    	(0, 2)	2
+    	(1, 3)	1
+    	(2, 0)	2
+    	(2, 3)	3
 
     >>> dist_matrix, predecessors = bellman_ford(csgraph=graph, directed=False, indices=0, return_predecessors=True)
     >>> dist_matrix
@@ -1253,11 +1265,14 @@ def johnson(csgraph, directed=True, indices=None,
     ... ]
     >>> graph = csr_matrix(graph)
     >>> print(graph)
-      (np.int32(0), np.int32(1))	1
-      (np.int32(0), np.int32(2))	2
-      (np.int32(1), np.int32(3))	1
-      (np.int32(2), np.int32(0))	2
-      (np.int32(2), np.int32(3))	3
+    <Compressed Sparse Row sparse matrix of dtype 'int64'
+    	with 5 stored elements and shape (4, 4)>
+    	Coords	Values
+    	(0, 1)	1
+    	(0, 2)	2
+    	(1, 3)	1
+    	(2, 0)	2
+    	(2, 3)	3
 
     >>> dist_matrix, predecessors = johnson(csgraph=graph, directed=False, indices=0, return_predecessors=True)
     >>> dist_matrix
@@ -1783,11 +1798,14 @@ def yen(
     ... ]
     >>> graph = csr_matrix(graph)
     >>> print(graph)
-      (np.int32(0), np.int32(1))	1
-      (np.int32(0), np.int32(2))	2
-      (np.int32(1), np.int32(3))	1
-      (np.int32(2), np.int32(0))	2
-      (np.int32(2), np.int32(3))	3
+    <Compressed Sparse Row sparse matrix of dtype 'int64'
+    	with 5 stored elements and shape (4, 4)>
+    	Coords	Values
+    	(0, 1)	1
+    	(0, 2)	2
+    	(1, 3)	1
+    	(2, 0)	2
+    	(2, 3)	3
 
     >>> dist_array, predecessors = yen(csgraph=graph, source=0, sink=3, K=2,
     ...                                directed=False, return_predecessors=True)

--- a/scipy/sparse/csgraph/_tools.pyx
+++ b/scipy/sparse/csgraph/_tools.pyx
@@ -453,10 +453,13 @@ def reconstruct_path(csgraph, predecessors, directed=True):
     ... ]
     >>> graph = csr_matrix(graph)
     >>> print(graph)
-       (np.int32(0), np.int32(1))	1
-       (np.int32(0), np.int32(2))	2
-       (np.int32(1), np.int32(3))	1
-       (np.int32(2), np.int32(3))	3
+    <Compressed Sparse Row sparse matrix of dtype 'int64'
+    	with 4 stored elements and shape (4, 4)>
+    	Coords	Values
+    	(0, 1)	1
+    	(0, 2)	2
+    	(1, 3)	1
+    	(2, 3)	3
 
     >>> pred = np.array([-9999, 0, 0, 1], dtype=np.int32)
 
@@ -570,10 +573,13 @@ def construct_dist_matrix(graph,
     ... ]
     >>> graph = csr_matrix(graph)
     >>> print(graph)
-       (np.int32(0), np.int32(1))	1
-       (np.int32(0), np.int32(2))	2
-       (np.int32(1), np.int32(3))	1
-       (np.int32(2), np.int32(3))	3
+    <Compressed Sparse Row sparse matrix of dtype 'int64'
+    	with 4 stored elements and shape (4, 4)>
+    	Coords	Values
+    	(0, 1)	1
+    	(0, 2)	2
+    	(1, 3)	1
+    	(2, 3)	3
 
     >>> pred = np.array([[-9999, 0, 0, 2],
     ...                  [1, -9999, 0, 1],

--- a/scipy/sparse/csgraph/_traversal.pyx
+++ b/scipy/sparse/csgraph/_traversal.pyx
@@ -75,10 +75,13 @@ def connected_components(csgraph, directed=True, connection='weak',
     ... ]
     >>> graph = csr_matrix(graph)
     >>> print(graph)
-      (np.int32(0), np.int32(1))	1
-      (np.int32(0), np.int32(2))	1
-      (np.int32(1), np.int32(2))	1
-      (np.int32(3), np.int32(4))	1
+    <Compressed Sparse Row sparse matrix of dtype 'int64'
+    	with 4 stored elements and shape (5, 5)>
+    	Coords	Values
+    	(0, 1)	1
+    	(0, 2)	1
+    	(1, 2)	1
+    	(3, 4)	1
 
     >>> n_components, labels = connected_components(csgraph=graph, directed=False, return_labels=True)
     >>> n_components
@@ -332,11 +335,14 @@ cpdef breadth_first_order(csgraph, i_start,
     ... ]
     >>> graph = csr_matrix(graph)
     >>> print(graph)
-      (np.int32(0), np.int32(1))	1
-      (np.int32(0), np.int32(2))	2
-      (np.int32(1), np.int32(3))	1
-      (np.int32(2), np.int32(0))	2
-      (np.int32(2), np.int32(3))	3
+    <Compressed Sparse Row sparse matrix of dtype 'int64'
+    	with 5 stored elements and shape (4, 4)>
+    	Coords	Values
+    	(0, 1)	1
+    	(0, 2)	2
+    	(1, 3)	1
+    	(2, 0)	2
+    	(2, 3)	3
 
     >>> breadth_first_order(graph,0)
     (array([0, 1, 2, 3], dtype=int32), array([-9999,     0,     0,     1], dtype=int32))
@@ -538,11 +544,14 @@ cpdef depth_first_order(csgraph, i_start,
     ... ]
     >>> graph = csr_matrix(graph)
     >>> print(graph)
-    (np.int32(0), np.int32(1))	1
-    (np.int32(0), np.int32(2))	2
-    (np.int32(1), np.int32(3))	1
-    (np.int32(2), np.int32(0))	2
-    (np.int32(2), np.int32(3))	3
+    <Compressed Sparse Row sparse matrix of dtype 'int64'
+    	with 5 stored elements and shape (4, 4)>
+    	Coords	Values
+    	(0, 1)	1
+    	(0, 2)	2
+    	(1, 3)	1
+    	(2, 0)	2
+    	(2, 3)	3
 
     >>> depth_first_order(graph,0)
     (array([0, 1, 3, 2], dtype=int32), array([-9999,     0,     0,     1], dtype=int32))

